packages/notebook-extension/src/index.ts

```diff
@@ -1572,6 +1581,70 @@ function activateNotebookHandler(
   return tracker;
 }
 
+/**
+ * Activate the completer service for notebook.
+ */
+function activateNotebookCompleterService(
+  app: JupyterFrontEnd,
+  notebooks: INotebookTracker,
+  manager?: ICompletionProviderManager
+): void {
+  if (!manager) {
+    return;
+  }
+  const invokeNotebook = 'completer:invoke-notebook';
+  const selectNotebook = 'completer:select-notebook';
+  app.commands.addCommand(invokeNotebook, {
+    execute: args => {
+      const panel = notebooks.currentWidget;
+      if (panel && panel.content.activeCell?.model.type === 'code') {
+        manager.invoke(panel.id);
+      }
+    }
+  });
+
+  app.commands.addCommand(selectNotebook, {
+    execute: () => {
+      const id = notebooks.currentWidget && notebooks.currentWidget.id;
+
+      if (id) {
+        return manager.select(id);
+      }
+    }
+  });
+
+  app.commands.addKeyBinding({
+    command: selectNotebook,
+    keys: ['Enter'],
+    selector: '.jp-Notebook .jp-mod-completer-active'
+  });
+
+  notebooks.widgetAdded.connect(async (_, notebook) => {
+    const completerContext = {
+      editor: notebook.content.activeCell?.editor ?? null,
+      session: notebook.sessionContext.session,
+      widget: notebook
+    };
+    await manager.attachWidget(notebook, completerContext);
+    notebook.content.activeCellChanged.connect((_, cell) => {
+      const newCompleterContext = {
+        editor: cell.editor,
+        session: notebook.sessionContext.session,
+        widget: notebook
+      };
+      manager.updateHandler(notebook.id, newCompleterContext);
```

Comment:
Same inconsistency as for `attachWidget`

Reply:
I merged `attachWidget` into  `updateHandler` with simplified signature.